src/dev/flang/fe/FrontEndOptions.java

```diff
@@ -133,7 +133,8 @@ public FrontEndOptions(int verbose,
                          List<String> sourceDirs,
                          boolean readStdin,
                          String main,
-                         boolean loadSources)
+                         boolean loadSources,
+                         boolean preconditionForNoInput)
```

Comment:
No, this was not what I meant!

Try not to add complexity, but try to understand why a condition is there and refine it.
In FrontEnd, I see that readStdin/main are needed iff loadSources is true, so a solution might be to add !loadSources || in front of the precondition.

Alternatively, you could remove the precondition here and add it somewhere, e.g., to the constructor of FrontEnd, but I prefer to check this as early as possible.

Thinking about it more: why is readStdin true at all? I do not think that fzjava should ever read from stdin. How do you run into this?